### PR TITLE
fix #144: indicate the change of license

### LINKED ISSUE #144: Releasing ANTLR Kotlin _also_ under the BSD-3 clause
Dear @lppedd @drieks @phyrian @exaV @OldWartHog @brianguertin @atsushieno @tianyu @tim-patterson @alchitry @oleg-zinovev @ianparkinson @garyttierney @frett @wilkomanger @unosviluppatore

I am writing about your contributions to ANTLR Kotlin. A project that I originally started in 2017 and that benefitted from your help.

ANTLR Kotlin is released under the Apache License 2. 

However a team designing ANTLR 5 is considering including components of ANTLR Kotlin in their project. It could end up being used only for experiments, or, as I hope, became the basis of ANTLR 5. The reason for this choice is that the ANTLR Kotlin runtime can be compiled to WASM.

I think it would be an amazing possibility to contribute to the new generation of ANTLR and make our work more valuable. For that the team behind ANTLR 5 requested if you would consider granting the license to your work _also_ under the BSD-3 license. The differences between the two licenses seem minimal to me. What I am proposing is one of these alternatives.

So I am studying how to satisfy this request, while letting users of ANTLR Kotlin keep using future releases of ANTLR Kotlin under the Apache License 2, if they wish to do so. So there would be no changes for them.

Based on your answers there are three possible scenarios:

# Scenario 1: ANTLR Kotlin under both licenses

If we get the approval from all the authors (or the authors of 99%+ of the commits) then we will:
- Provide future versions of ANTLR Kotlin under both the Apache License 2 and the BSD-3 License. The users will decide which one they want to apply
- The ANTLR 5 team could use ANTLR Kotlin and benefits from future fixes
- Remove from the repository the contributions of the authors who do not agree to releasing their work _also_ under the BSD-3 License

# Scenario 2: ANTLR Kotlin as-is, plus "revisited fork"

If we get the approval from most of the authors (or the authors of 95%+ of the commits) then we will:
- Let the ANTLR Kotlin repository unchanged and provide _only_ after the Apache License 2, as it is the case now
- We will create a fork version of ANTLR Kotlin and remove from it the contributions of the authors who do not agree to releasing their work _also_ under the BSD-3 License. This fork will be _only_ under the BSD-3 license
- The ANTLR 5 team could use the forked version of ANTLR Kotlin

# Scenario 3: We do not proceed

If we do not the approval from most of the authors then we will drop this possibility:
- The ANTLR Kotlin repository will provide as it is now
- The ANTLR 5 team will rewrite the runtime on their own, without reusing any code from the ANTLR Kotlin project

---

# What do I need from you?

To answer this issue stating that you grant me the possibility to release your contributions to ANTLR Kotlin _also_ under the BSD-3 License, or alternatively to send me an email about that.

I am sorry to disturb you with this boring discussion about licenses and I am thankful for your contributions to the ANTLR Kotlin project. It has been providing value to different adopters during these years, and the fact we have the possibility to provide even more value is a great thing that you made possible.

Cheers,
Federico

### PR DESCRIPTION
**Pull request by ftomassetti**
Fix #144 

It specifies that the project is available under both Apache License 2.0 or BSD 3-Clause.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -164,3 +165,9 @@ You can see the complete list on GitHub, but here we list those who contributed
 - Patrick Del Conte [@exaV](https://github.com/exaV)
 
 [1]: https://plugins.gradle.org/plugin/com.strumenta.antlr-kotlin
+
+## License
+
+Users of this project can choose which license to apply: either the Apache License 2 or the BSD 3-Clause License.
```
Comment (lppedd) [inline README.md:171]:
```suggestion
Consumers of this project may choose the most appropriate license: either the Apache License 2.0, or the 3-Clause BSD License.
```

File: README.md [hunk 2]
```diff
@@ -164,3 +165,9 @@ You can see the complete list on GitHub, but here we list those who contributed
...
+
+The licenses are alternatively.
```
Comment (lppedd) [inline README.md:173]: Don't think this is needed, as the above sentence already states a consumer have the choice.

File: README.md [hunk 3]
```diff
@@ -23,7 +24,7 @@ The runtime and target's template are up-to-date with mainstream commit: `master
 
 ## Kotlin Target
 
-The Kotlin target is a plugin for the ANTLR generator.  
+The Kotlin target is a plugin for the ANTLR generator.
```
Comment (lppedd) [inline README.md:27]: I think the formatter got rid of the double whitespace automatically.

File: README.md (hunk 2 above)
Comment (ftomassetti) [reply README.md]: Removing it

File: README.md (hunk 3 above)
Comment (ftomassetti) [reply README.md:27]: I am not sure if you think it should be re-added

Comment (lppedd) [reply README.md:27]: @ftomassetti I'd add the removed double spaces back (there are two edits for this case). Just for better formatting in the rendered markdown.

Comment (ftomassetti) [reply README.md:27]: IDEA seems to remove those spaces on save, making it very inconvenient to keep them. I would instead add an empty line between this sentence and the next one.

Comment (lppedd) [reply README.md:27]: @ftomassetti it's ok. But FYI, it's caused by the EditorConfig settings. There is trim trailing ws option that is on save (and then you go to commit).

What I do usually is set that option to false, commit the whitespaces, and set it back to true.

Comment (ftomassetti) [reply README.md:27]: I think we can consider that in a separate issue, but for now I will merge this (squashing the commit), merge this and then reviewed the pending PRs

**Comment by lppedd** [comment]
Not a big deal for the whitespaces, we can get back to it in any PR that changes the readme again.

**Pull request merged**